apps/dapp-console/app/settings/contracts/DeployedApp.tsx

```diff
@@ -0,0 +1,129 @@
+'use client'
+
+import {
+  Button,
+  Card,
+  CardContent,
+  CardHeader,
+  CardTitle,
+} from '@eth-optimism/ui-components'
+
+import { AddContractFlow } from '@/app/settings/contracts/AddContractFlow'
+import { RiMoreLine } from '@remixicon/react'
+import { EditAppDialog } from '@/app/settings/contracts/EditAppDialog'
+import { useCallback, useState } from 'react'
+import { DeployedApp as ApiDeployedApp, Contract } from '@/app/types/api'
+import { DeployedAppContract } from '@/app/settings/contracts/DeployedAppContract'
+import { apiClient } from '@/app/helpers/apiClient'
+import { StartVerificationDialog } from '@/app/settings/contracts/StartVerificationDialog'
+
+export type DeployedAppProps = {
+  app: ApiDeployedApp
+}
+
+export const DeployedApp = ({ app }: DeployedAppProps) => {
+  const [contracts, setContracts] = useState(app.contracts)
+  const [contractToVerifiy, setContractToVerify] = useState<
+    Contract | undefined
+  >()
+  const [hasDraftContract, setHasDraftContract] = useState(
+    app.contracts.length === 0,
+  )
+  const [isVerifyDialogOpen, setVerifyDialogOpen] = useState(false)
+  const [appName, setAppName] = useState<string>(app.name)
+
+  const { refetch: fetchContracts } =
+    apiClient.Contracts.listContractsForApp.useQuery(
+      { appId: app.id },
+      { enabled: false },
+    )
+
+  const handleUpdateAppName = useCallback(
+    (updatedName: string) => {
+      setAppName(updatedName)
+    },
+    [setAppName],
+  )
+
+  const handleStartVerification = useCallback(
+    async (contract: Contract) => {
+      setContractToVerify(contract)
+      setVerifyDialogOpen(true)
+
+      const { data: contracts } = await fetchContracts()
+      setContracts(contracts as Contract[])
```

Comment:
how come this has to be casted?

Reply:
i think it's because of the enum/string difference